src/WebJobs.Script.WebHost/HostJsonConfigProvider.cs

```diff
@@ -0,0 +1,138 @@
+        {
+            if (e.NewHost != null)
+            {
+                ExtensionOptions = GetExtensionOptions(_extensionOptionTypes, e.NewHost.Services);
+                ConcurrencyOption = GetConcurrencyOption(e.NewHost.Services);
+            }
+        }
+
+        private Dictionary<string, JObject> GetExtensionOptions(List<Type> extensionOptionTypes, IServiceProvider serviceProvider)
+        {
+            var configuration = serviceProvider.GetService<IConfiguration>();
+            // for each of the extension options types identified, create a bound instance
+            // and format to JObject
+            Dictionary<string, JObject> result = new Dictionary<string, JObject>();
+            foreach (Type optionsType in extensionOptionTypes)
+            {
+                // create the IOptions<T> type
+                Type optionsWrapperType = typeof(IOptions<>).MakeGenericType(optionsType);
+                object optionsWrapper = serviceProvider.GetService(optionsWrapperType);
+                PropertyInfo valueProperty = optionsWrapperType.GetProperty("Value");
+
+                // create the instance - this will cause configuration binding
+                object options = valueProperty.GetValue(optionsWrapper);
+
+                // get the section name from Extension attribute of IExtensionConfigProvider class
+                Type webJobsExtensionOptionsConfigurationType = typeof(IWebJobsExtensionOptionsConfiguration<>).MakeGenericType(optionsType);
+                var webJobsExtensionOptionsConfiguration = serviceProvider.GetService(webJobsExtensionOptionsConfigurationType);
+                PropertyInfo sectionName = webJobsExtensionOptionsConfigurationType.GetProperty("ExtensionInfo");
+                IExtensionInfo info = (IExtensionInfo)sectionName.GetValue(webJobsExtensionOptionsConfiguration);
+
+                string name = info.ConfigurationSectionName.CamelCaseString();
+
+                IOptionsFormatter optionsFormatter = options as IOptionsFormatter;
+                if (optionsFormatter != null)
+                {
+                    result.Add(name, JObject.Parse(optionsFormatter.Format()).ToCamelCase());
+                }
+                else
+                {
+                    // We don't support the extensions that doesn't implement IOptionsFormatter.
+
+                    // TODO : For Avoiding Breaking change of Durable Functions with Arc
+                    // Once Durable Functions supoorts IOptionsFormatter, remove this line
+                    if ("durableTask".Equals(name, StringComparison.OrdinalIgnoreCase))
+                    {
+                        result.Add(name, JObject.FromObject(options).ToCamelCase());
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        public JObject GetConcurrencyOption(IServiceProvider serviceProvider)
+        {
+            var configuration = serviceProvider.GetService<IConfiguration>();
```

Comment:
I don't understand why you're accessing options like this. You should just take ConcurrencyOptions via DI and let it resolve for you?

Reply:
Change into DI.